### DISCUSSION TITLE
How do I determine the output library programmatically?

### DISCUSSION THREAD

**Discussion by davesworking** [microsoft/vcpkg#29915] [Q&A]
Is there a way to determine the name of the output library, for any triplet/OS, programmatically before vcpkg even runs? If not, how can that be determined after it runs?

**Comment by dg0yt** [comment]
> Is there a way to determine the name of the output library, for any triplet/OS, programmatically before vcpkg even runs? 

No. In general, vcpkg doesn't know the names at all. This information is in the actual build systems.

> If not, how can that be determined after it runs?

It can be determined only from files installed by the actual build system, in particular generated CMake config or pkg-config files.
Many packages lack such files. 
Some vcpkg ports add "unofficial" CMake config by patching.
Some vcpkg ports instrument official CMake Find modules to correctly find the installed libraries, usually by specifying the set of candidate names and locations for searching the first match. 
Actually this is the general approach: inspect the `lib` directory.

Maybe it would be helpful to know what you want to do with the name of the output library...

**Reply by dg0yt** [reply]
FTR some ports don't install libraries at all, but may fail anyways.

**Comment by davesworking** [comment]
We would like to check for the output library's existence and fail the build if it isn't there.

**Reply by Osyotr** [reply]
A successful vcpkg install command leaves `vcpkg_installed/share/<port-name>/vcpkg-abi-info.txt`. Unsuccessful build leaves nothing.

**Reply by dg0yt** [reply]
I can't believe that  there is no proper way to pass build failure through the call stack. (And it is really unrelated to library names in the end...)
